web/src/main/kotlin/no/nav/su/se/bakover/web/routes/vilkår/utenlandsopphold/LeggTilUtenlandsoppholdRoute.kt

```diff
@@ -0,0 +1,182 @@
+                            )
+                    }
+                }
+            }
+        }
+    }
+}
+
+internal data class UtenlandsoppholdBody(
+    val vurderinger: List<UtenlandsoppholdVurderingBody>,
+) {
+    fun tilLeggTilUtenlandsoppholdrequest(behandlingId: UUID): Either<Resultat, LeggTilUtenlandsoppholdRequest> =
+        if (vurderinger.size != 1) BadRequest.errorJson(
+            "Flere vurderingsperioder ble sendt inn. Forventet kun 1",
+            "flere_perioder_ble_Sendt_inn",
+        ).left() else LeggTilUtenlandsoppholdRequest(
+            behandlingId = behandlingId,
+            periode = vurderinger.first().periode.toPeriodeOrResultat().getOrHandle { return it.left() },
+            status = vurderinger.first().status,
+        ).right()
+
+    fun tilLeggTilFlereUtenlandsoppholdRequest(revurderingId: UUID): Either<Resultat, LeggTilFlereUtenlandsoppholdRequest> =
+        LeggTilFlereUtenlandsoppholdRequest(
+            behandlingId = revurderingId,
+            request = Nel.fromListUnsafe(
+                vurderinger.map {
+                    LeggTilUtenlandsoppholdRequest(
+                        behandlingId = revurderingId,
+                        periode = it.periode.toPeriodeOrResultat().getOrHandle { return it.left() },
+                        status = it.status,
+                    )
+                },
+            ),
+        ).right()
+}
+
+internal data class UtenlandsoppholdVurderingBody(
+    val periode: PeriodeJson,
+    val status: UtenlandsoppholdStatus,
+)
+
+internal fun SøknadsbehandlingService.KunneIkkeLeggeTilUtenlandsopphold.tilResultat(): Resultat {
+    return when (this) {
+        SøknadsbehandlingService.KunneIkkeLeggeTilUtenlandsopphold.FantIkkeBehandling -> {
+            Feilresponser.fantIkkeBehandling
+        }
+
+        is SøknadsbehandlingService.KunneIkkeLeggeTilUtenlandsopphold.UgyldigTilstand -> {
+            Feilresponser.ugyldigTilstand(fra = this.fra, til = this.til)
+        }
+
+        SøknadsbehandlingService.KunneIkkeLeggeTilUtenlandsopphold.VurderingsperiodeUtenforBehandlingsperiode -> {
+            Feilresponser.utenforBehandlingsperioden
+        }
+
+        SøknadsbehandlingService.KunneIkkeLeggeTilUtenlandsopphold.AlleVurderingsperioderMåHaSammeResultat -> {
+            Feilresponser.alleResultaterMåVæreLike
+        }
+
+        SøknadsbehandlingService.KunneIkkeLeggeTilUtenlandsopphold.MåInneholdeKunEnVurderingsperiode -> {
+            Feilresponser.måInnheholdeKunEnVurderingsperiode
+        }
+
+        SøknadsbehandlingService.KunneIkkeLeggeTilUtenlandsopphold.MåVurdereHelePerioden -> {
+            Feilresponser.måVurdereHelePerioden
+        }
+    }
+}
+
+internal fun KunneIkkeLeggeTilUtenlandsopphold.tilResultat(): Resultat {
+    return when (this) {
+        KunneIkkeLeggeTilUtenlandsopphold.FantIkkeBehandling -> {
+            HttpStatusCode.NotFound.errorJson("Fant ikke revurdering", "fant_ikke_revurdering")
```

Comment:
Tenker vi ikke trenger en egen for revurdering og søknadsbehandling. Burde holde med en fant_ikke_behandling (som dekker begge)